dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java

```diff
@@ -50,613 +50,620 @@
-		return folderFactory.findFoldersByHost(host);
-	}
-
-	@CloseDBIfOpened
-	public  List<Link> getLinks(final Folder parent, final boolean working,
-								final boolean deleted, final User user,
-								final boolean respectFrontEndPermissions)
-								throws DotStateException, DotDataException, DotSecurityException {
+    @CloseDBIfOpened
+    public Folder findFolderByPath(final String path, final Host host,
+                                   final User user, final boolean respectFrontEndPermissions) throws DotStateException,
+            DotDataException, DotSecurityException {
+
+        final Folder folder = folderFactory.findFolderByPath(path, host);
+
+        if (folder != null && InodeUtils.isSet(folder.getInode()) &&
+                !permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)) {
+
+            // SYSTEM_FOLDER means if the user has permissions to the host, then they can see host.com/
+            if(FolderAPI.SYSTEM_FOLDER.equals(folder.getInode())) {
+                if(!Host.SYSTEM_HOST.equals(host.getIdentifier())){
+                    if(!permissionAPI.doesUserHavePermission(host, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)) {
+                        throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to read folder " + folder.getPath());
+                    }
+                }
+            }
+
+        }
+
+        return folder;
+    }
+
+
+    public Folder findFolderByPath(String path, String hostid, User user, boolean respectFrontEndPermissions) throws DotStateException,
+            DotDataException, DotSecurityException {
+        if(user==null) user = APILocator.getUserAPI().getSystemUser();
+        Host host = APILocator.getHostAPI().find(hostid, user, false);
+        return findFolderByPath(path,host,user, respectFrontEndPermissions);
+    }
+
+    @WrapInTransaction
+    public boolean renameFolder(final Folder folder, final String newName,
+                                final User user, final boolean respectFrontEndPermissions) throws DotDataException,
+            DotSecurityException {
+
+        boolean renamed = false;
+
+        if (!permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_EDIT, user, respectFrontEndPermissions)) {
+            throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to edit folder" + folder.getPath());
+        }
+
+        try {
+
+            renamed = folderFactory.renameFolder(folder, newName, user, respectFrontEndPermissions);
+
+            return renamed;
+        } catch (Exception e) {
+
+            throw new DotDataException(e.getMessage(),e);
+        }
+    }
+
+    @CloseDBIfOpened
+    public Folder findParentFolder(final Treeable asset, final User user, final boolean respectFrontEndPermissions) throws DotIdentifierStateException,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 151 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)